piecedirectory/couchbase_test.go

```diff
@@ -0,0 +1,255 @@
+package piecedirectory
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/couchbase/gocb/v2"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	dockercl "github.com/docker/docker/client"
+	"github.com/docker/go-connections/nat"
+	"github.com/filecoin-project/boostd-data/couchbase"
+	logging "github.com/ipfs/go-log/v2"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/net/context"
+)
+
+var tlog = logging.Logger("pdtest")
+
+func init() {
+	logging.SetLogLevel("pdtest", "debug")
+}
+
+func setupCouchbase(t *testing.T) {
+	ctx := context.Background()
+	cli, err := dockercl.NewClientWithOpts(dockercl.FromEnv)
+	require.NoError(t, err)
+
+	imageName := "couchbase"
+	out, err := cli.ImagePull(ctx, imageName, types.ImagePullOptions{})
+	require.NoError(t, err)
+
+	_, err = io.Copy(os.Stdout, out)
+	require.NoError(t, err)
+
+	tlog.Info("couchbase docker container create...")
+	resp, err := cli.ContainerCreate(ctx, &container.Config{
+		Image: imageName,
+		ExposedPorts: nat.PortSet{
+			"8091":  struct{}{},
+			"8092":  struct{}{},
+			"8093":  struct{}{},
+			"8094":  struct{}{},
+			"8095":  struct{}{},
+			"8096":  struct{}{},
+			"11210": struct{}{},
+			"11211": struct{}{},
+		},
+	}, &container.HostConfig{
+		PortBindings: map[nat.Port][]nat.PortBinding{
```

Comment:
how much harder would it be to make these not fixed ports on the host so that multiple runs won't conflict?

Reply:
all tests should really use a single instance of couchbase, we shouldn't start and stop it in between tests, as that's a quite slow process...

if cleanup is not done properly, i am not sure starting a bunch of instances on a different port is really nice either, i think we'd rather just get a conflict.

Reply:
if we want to be really clean, then it makes sense to use custom and non-default couchbase ports, but thats really not a pain point right now.